source/IAccessibleHandler/__init__.py

```diff
@@ -810,7 +810,9 @@ def _get_role(self):
 		return controlTypes.Role.PANE
 
 	def event_gainFocus(self):
-		super(SecureDesktopNVDAObject, self).event_gainFocus()
+		from speech.speech import cancelSpeech
+		cancelSpeech()

```

Comment:
Please add a comment to explain why this is required here.

Reply:
It would be good to mention this was previously handled by the foreground event (to a `Desktop(Window)` ). The foreground event is now suppressed for security reasons.

Reply:
this has been done

Reply:
>Cancelling speech was previously handled by a foreground event on SecureDesktopNVDAObject.

Is this misleading? The foreground object that was set was previously via `api.getDesktopObject().objectInForeground()`

If I recall correctly that was a `Desktop` object (I.E. for the logged in user, not the system / secure desktop)